frontend/src/main/components/HelpRequest/HelpRequestForm.js

```diff
@@ -0,0 +1,176 @@
+import { Button, Form, Row, Col } from 'react-bootstrap';
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+
+function HelpRequestForm({ initialContents, submitAction, buttonLabel = "Create" }) {
+
+    // Stryker disable all
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+    } = useForm(
+        { defaultValues: initialContents || {}, }
+    );
+    // Stryker restore all
+
+    const navigate = useNavigate();
+
+    // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+    // Note that even this complex regex may still need some tweaks
+
+    // Stryker disable next-line Regex
+    const isodate_regex = /(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d)/i;
+
+    // Stryker disable next-line all
+    const email_regex = /[\w.]+@([\w]+\.)+[\w-]{2,4}/i;
+    const teamId_regex = /[smfw]\d\d-([123456789]|10|11|12)(am|pm)-[1234]$/;
+    const solved_regex = /(true|false)/i;
```

Comment:
```suggestion
    const solved_regex = /^(true|false)$/i;
```
`^` followed by `$` ensures that no other characters appear before or after true/false. Also (Im not sure) are booleans case sensitive (i.e. fAlSe should not work)? If so, `/i` should be removed as well.

Reply:
Thank you! I have changed it.